libc/src/__support/FPUtil/ManipulationFunctions.h

```diff
@@ -169,8 +171,49 @@ LIBC_INLINE T nextafter(T from, T to) {
     int_val = (to_bits.uintval() & sign_mask) + UIntType(1);
   }
 
+  UIntType exponent_bits = int_val & FloatProperties<T>::EXPONENT_MASK;
+  if (exponent_bits == UIntType(0))
+    raise_except_if_required(FE_UNDERFLOW | FE_INEXACT);
+  else if (exponent_bits == FloatProperties<T>::EXPONENT_MASK)
+    raise_except_if_required(FE_OVERFLOW | FE_INEXACT);
+
+  return cpp::bit_cast<T>(int_val);
+}
+
+template <typename T, cpp::enable_if_t<cpp::is_floating_point_v<T>, int> = 0>
+LIBC_INLINE T nexttoward(T from, long double to) {
```

Comment:
For free function, I personally prefer to use SFINAE on the returning type `LIBC_INLINE cpp::enable_if_t<cpp::is_floating_point_v<T>, T> nexttoward(...)`, so that it frees the second template parameter to be used if needed.

Reply:
Essentially, when we do SFINAE with `template<T, U = enable_if_t<..., void>>`, we do not intend to let user use the second template parameter, but nothing prevents user code to be `func<T, U>(...)`, and it might silently get wrong function.  If all users do is to call `func(...)` or `func<T>` then it doesn't matter.